### ISSUE TITLE
Optimize the api response for territories

### ISSUE THREAD

**Issue by amaurymartiny** [chronoscio/backend#29]
Right now, if I am on the map on the frontend, it makes an api call `/territories?bounds=...`, and if I slightly move the map, the frontend makes another call `/territories?bounds=...`, with slightly different bounds. 99% of the cases the 2 responses will be identical.

An idea is to pass `/territories?bounds=...&territories=[id1,id2,id3]`, where `[id1,id2,id3]` are all Territory ids that the frontend already has. The backend then does its normal GIS intersection calculations, and if the id of newly calculated territories is in `[id1,id2,id3]`, then strip them out of the response.

**Comment by dwaxe** [comment]
I noticed this problem when I made the first version of this backend -- the shape downloads are extremely slow. Even if we cache territories upon map movement, the initial load will be unbearably slow.

To deal with that, we might want to consider geobuf (https://github.com/mapbox/geobuf)/something similar + compressing the response.

**Comment by amaurymartiny** [comment]
@dwaxe Separated your optimization in another issue, so we can track them independently: #38.

**Comment by wtokumaru** [comment]
Would it not be possible to cache entire maps via some separate database of caches?

**Comment by dwaxe** [comment]
@wtokumaru yes, in the future I'm imagining we will come up with some caching scheme that Will make our API calls faster and more efficient.
